src/transformers/models/arcee/modular_arcee.py

```diff
@@ -0,0 +1,337 @@
+            head_dim=head_dim,
+            **kwargs,
+        )
+
+        # Validate the correctness of rotary position embeddings parameters using Arcee's custom validation
+        # BC: if there is a 'type' field, copy it to 'rope_type'.
+        if self.rope_scaling is not None and "type" in self.rope_scaling:
+            self.rope_scaling["rope_type"] = self.rope_scaling["type"]
+        rope_config_validation(self)
+
+
+class ArceeRMSNorm(LlamaRMSNorm):
+    """ArceeRMSNorm is identical to LlamaRMSNorm"""
+
+    pass
+
+
+class ArceeMLP(nn.Module):
+    """Arcee MLP with configurable activation function (typically relu2)"""
+
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.hidden_size = config.hidden_size
+        self.intermediate_size = config.intermediate_size
+        self.up_proj = nn.Linear(self.hidden_size, self.intermediate_size, bias=config.mlp_bias)
+        self.down_proj = nn.Linear(self.intermediate_size, self.hidden_size, bias=config.mlp_bias)
+        self.act_fn = ACT2FN[config.hidden_act]
+
+    def forward(self, x):
+        down_proj = self.down_proj(self.act_fn(self.up_proj(x)))
+        return down_proj
+
+
+class ArceeAttention(LlamaAttention):
+    """Multi-headed attention for Arcee - identical to Llama attention"""
+
+    pass
+
+
+class ArceeDecoderLayer(LlamaDecoderLayer):
+    """Arcee decoder layer with custom MLP"""
+
+    def __init__(self, config: ArceeConfig, layer_idx: int):
+        super().__init__(config, layer_idx)
+        # Replace the Llama MLP with Arcee MLP to use the correct activation
+        self.mlp = ArceeMLP(config)
+
+
+class ArceePreTrainedModel(LlamaPreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = ArceeConfig
+
+
+class ArceeModel(LlamaModel):
+    """
+    Transformer decoder consisting of *config.num_hidden_layers* layers. Each layer is a [`ArceeDecoderLayer`]
+
+    Args:
+        config: ArceeConfig
+    """
+
+    def __init__(self, config: ArceeConfig):
+        super().__init__(config)
+        # The parent init handles most setup, we just need to ensure our decoder layers are used
+        self.layers = nn.ModuleList(
+            [ArceeDecoderLayer(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
+        )
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class ArceeForCausalLM(LlamaForCausalLM):
+    """Arcee Model transformer with a language modeling head on top (linear layer with weights tied to the input embeddings)."""
+
+    def __init__(self, config):
+        # We need to set config_class before calling super().__init__
+        self.config_class = ArceeConfig
+        super().__init__(config)
+        self.model = ArceeModel(config)
+        # Initialize weights and apply final processing
+        self.post_init()
```

Comment:
This should be sufficient!

```suggestion
class ArceeForCausalLM(LlamaForCausalLM):
    pass
```